src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -602,10 +602,28 @@ private OpenApiParameter GenerateParameterWithoutFilter(
                 In = location,
                 Required = isRequired,
                 Schema = schema,
-                Description = description
+                Description = description,
+                Style = GetParameterStyle(type, apiParameter.Source)
             };
         }
 
+        private static ParameterStyle? GetParameterStyle(Type type, BindingSource source)
+        {
+            return source == BindingSource.Query && IsKeyValueStringsPairCollection(type) ? ParameterStyle.DeepObject : null;
+
+            bool IsKeyValueStringsPairCollection(Type t)
+            {
+                if (!t.IsGenericType)
+                {
+                    return false;
+                }
+
+                return typeof(IDictionary<string, string>).IsAssignableFrom(t) ||
+                       typeof(IReadOnlyDictionary<string, string>).IsAssignableFrom(t) ||
+                       typeof(IEnumerable<KeyValuePair<string, string>>).IsAssignableFrom(t);
```

Comment:
The first two both inherit from the third, so either we only want just the first two or it's just the third one.

Reply:
thanks, I think it's OK to support `IEnumerable<KeyValuePair<string, string>>`